### PR TITLE
seperate compiler from framework tests

### PR DESCRIPTION
**Pull request by hkollmann**

### CODE REVIEW

**Comment by cboulanger** [comment]
Since the changes involve so many files, can you briefly explain what this does in some more detail?

File: test/framework/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (johnspackman) [inline test/framework/.vscode/launch.json:1]: is this file meant to be included?  If it is, please can it have comments in english?

Comment (cboulanger) [reply test/framework/.vscode/launch.json:1]: IDE files should never be included. Maybe add to `.gitignore`?

Comment (hkollmann) [reply test/framework/.vscode/launch.json]: You are right

Comment (hkollmann) [reply test/framework/.vscode/launch.json]: Deleted and added .gitignore

**Review by cboulanger — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Comment by cboulanger** [comment]
Thank you @hkollmann !

**Pull request merged**